doc/environment.yml

```diff
@@ -28,3 +37,4 @@ dependencies:
     - nbsphinx
     - jupyter_client
     - pickleshare
+    - pre-commit
```

Comment:
We do not need pre-commit for the docs, do we?

Reply:
> We do not need pre-commit for the docs, do we?

The pre-commit bot runs on all PRs regardless of the changes. I think we should promote users fixing their formatting issues rather than waiting for the bot to try and fail

Reply:
> It only does once you have installed it! But I think you are making a great point about fixing formatting. But lets parse the two different scenarios here:
> 
> * What is needed to exclusively build the docs
> * What is needed to locally develop the docs

Since the docs has more than just `.rst` files -` ipynb` notebooks as well, the local development and build would be the same? or maybe im missing the point of splitting build vs deployment

> Can we leave this env file as is and instruct the user to install the cloned repo with `pip install ".[dev]" (where we would add the pre-commit dependency)?

In order to do that, we need all the dependencies needed to compile docs notebooks to be available when we do `pip install .[dev]`. I can update that, if thats how we want to do build vs deployment. Then we can remove pre-commit from `doc/environment.yml` as well

> Also are we sure that all the docs are actually checked by the pre-commit hooks?
changes like `trailing-whitespace` and `end-of-file-fixer` is run on all file types. The remaining is code related so some notebooks. And in the future if we add more notebooks, then those also need pre-commit linting

> Writing this I realize that this should probably be separated out into another PR to keep this already large PR more focused and keep it moving? We can continue the discussion over there.
When it comes to dependencies, the `doc/transforms.ipynb` did need to add a few more installations before it started working. Let me know what other aspect you want to split from this PR

Reply:
> In order to do that, we need all the dependencies needed to compile docs notebooks to be available when we do pip install .[dev]. I can update that, if thats how we want to do build vs deployment. Then we can remove pre-commit from doc/environment.yml as well

Well that would be ideal, but is difficult due to our non-python dependencies. My primary goal here is to *not add anything we do not need for the pure CI doc build*! This is a slightly different use case from modifiying the docs locally, for which (in my mind) you need the doc build env + dev tools like pre-commit.

Reply:
> Then we can remove pre-commit from doc/environment.yml as well

Yes that would be great!

Reply:
> When it comes to dependencies, the doc/transforms.ipynb did need to add a few more installations before it started working. Let me know what other aspect you want to split from this PR

Whatever is needed to get the docs build on RTD to pass is totally fine here.